packages/sanity/src/core/form/studio/assetSourceMediaLibrary/hooks/useProvision.ts

```diff
@@ -0,0 +1,212 @@
+  const observable = useMemo(
+    () =>
+      getOrganizationIdFromProjectId(client, projectId).pipe(
+        switchMap((organizationId) => {
+          // console.log('Fetched Organization ID:', organizationId)
+          return getMediaLibrariesForOrganization(client, organizationId).pipe(
+            concatMap((mediaLibraryData) => {
+              if (mediaLibraryData.length > 0) {
+                const id = mediaLibraryData[0].id
+                if (onLibraryId) {
+                  onLibraryId(id)
+                }
+                return of({id, organizationId, status: 'active'} satisfies ProvisionResponse)
+              }
+              return createMediaLibrary(client, organizationId).pipe(
+                // eslint-disable-next-line max-nested-callbacks
+                concatMap((createdData) => {
+                  if (createdData && createdData.id) {
+                    return of({
+                      id: createdData.id,
+                      organizationId,
+                      status: 'provisioning',
+                    } satisfies ProvisionResponse)
+                  }
+                  throw new ProvisionError(
+                    'Failed to create Media Library',
+                    'Failed to create Media Library',
+                    'ERROR_FAILED_TO_CREATE_MEDIA_LIBRARY',
+                  )
+                }),
+                concatMap((libraryData) =>
+                  concat(
+                    of(libraryData), // Emit the initial response immediately
+                    interval(1000).pipe(
+                      concatMap(() => {
+                        // Poll for the media library status
+                        if (libraryData.id) {
+                          return requestMediaLibraryStatus(client, libraryData.id).pipe(
+                            concatMap((statusData) => {
+                              if (statusData.status === 'active') {
+                                if (onLibraryId) {
+                                  onLibraryId(statusData.id ?? '')
+                                }
+                              }
+                              return of(libraryData)
+                            }),
+                          )
+                        }
+                        return of(libraryData)
+                      }),
+                      takeWhile((pData) => pData?.status === 'provisioning', true),
+                    ),
+                  ),
+                ),
+              )
+            }),
+          )
+        }),
+        catchError((err) => {
+          if (err instanceof ProvisionError && onProvisionError !== undefined) {
+            return onProvisionError(err)
+          }
+          throw err
+        }),
+      ),
+    [client, projectId, onLibraryId, onProvisionError],
+  )
+
+  return (
+    useObservable(observable) ?? {
+      id: undefined,
+      organizationId: undefined,
+      status: 'loading',
+    }
+  )
```

Comment:
Use an initial value on the observable instead 
```suggestion
  return useObservable(observable)
```

Reply:
Yeah, the `useObservable` hook has a param for initial value. 👍